pkg/graph/schema.resolvers.go

```diff
@@ -2021,11 +2024,33 @@ func (r *mutationResolver) DeleteTRBRequestFundingSources(ctx context.Context, i
 
 // SetRolesForUserOnSystem is the resolver for the setRolesForUserOnSystem field.
 func (r *mutationResolver) SetRolesForUserOnSystem(ctx context.Context, input model.SetRolesForUserOnSystemInput) (*string, error) {
-	err := r.cedarCoreClient.SetRolesForUser(ctx, input.CedarSystemID, input.EuaUserID, input.DesiredRoleTypeIDs)
+	rs, err := r.cedarCoreClient.SetRolesForUser(ctx, input.CedarSystemID, input.EuaUserID, input.DesiredRoleTypeIDs)
 	if err != nil {
 		return nil, err
 	}
 
+	// Asyncronously send an email to the CEDAR team notifying them of the change
+	go func() {
+		// make a new context, or else the request will cancel when the parent context cancels
+		emailCtx := context.Background()
+		appcontext.WithLogger(emailCtx, appcontext.ZLogger(ctx)) // copy the logger over
```

Comment:
This doesn't work; `appcontext.WithLogger()` returns a new context, it doesn't change the existing context. You can test this by editing the local fake LDAP client in `pkg/local/cedar_ldap.go` and have `FetchUserInfo()` return an error immediately; the "failed to lookup user info for CEDAR notification email" error message won't get logged, and you'll see a "Logger not found on context." message in the `easi` container logs instead.

What you can do is combine these two lines into `emailCtx := appcontext.WithLogger(context.Background(), appcontext.ZLogger(ctx))`; then `emailCtx` should have a separate context, with the logger copied over.

Reply:
Thank you for catching this.

Not that it's an excuse, but this does remind me that I _thought_ we had something in the linter that required a statement like this to be invalid, and my (incorrect) code would have had to have been written as 
```go
		emailCtx := context.Background()
		_ = appcontext.WithLogger(emailCtx, appcontext.ZLogger(ctx)) // copy the logger over
```
to make it clear that the function DOES have a return, we're just ignoring it (rather than implicitly stating that)

Reply:
Either way - resolved in bd4ed24ea